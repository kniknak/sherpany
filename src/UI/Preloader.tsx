import React, {memo} from "react";
import Typography from "@material-ui/core/Typography";
import Toolbar from "@material-ui/core/Toolbar";

export const Preloader: React.FC = memo(({ children }) => {
    return (
        <Toolbar>
            <Typography
                align="center"
                style={{ flex: 1 }}
            >
                {children}
            </Typography>
        </Toolbar>
    );
});