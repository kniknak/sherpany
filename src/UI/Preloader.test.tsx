import React from "react";
import { shallow } from "enzyme";
import { Preloader } from "./Preloader";

describe("Preloader", () => {
    describe("Preloader", () => {
        it("renders", () => {
            const Foo = () => null
            const subject = shallow(<Preloader><Foo/></Preloader>);

            expect(subject).toMatchSnapshot();
        });
    });
});
