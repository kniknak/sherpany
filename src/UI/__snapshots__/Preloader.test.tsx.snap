// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Preloader Preloader renders 1`] = `
<WithStyles(ForwardRef(Toolbar))>
  <WithStyles(ForwardRef(Typography))
    align="center"
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <Foo />
  </WithStyles(ForwardRef(Typography))>
</WithStyles(ForwardRef(Toolbar))>
`;
